src/dinerl.erl

```diff
@@ -193,56 +194,79 @@ update_item(Table, Key, Options, Timeout) ->
     update_item(Table, Key, Options, [], Timeout).
 
 update_item(T, K, [], Acc, Timeout) ->
-    api(update_item, [{<<"TableName">>, T}, {<<"Key">>, K}|Acc], Timeout);
+    api(update_item, [{<<"TableName">>, T}, {<<"Key">>, K} | Acc], Timeout);
 update_item(T, K, [{update, AttributeUpdates}|Rest], Acc, Timeout) ->
-    update_item(T, K, Rest, [{<<"AttributeUpdates">>, attr_updates(AttributeUpdates, [])}|Acc], Timeout);
+    update_item(T, K, Rest, [{<<"AttributeUpdates">>, attr_updates(AttributeUpdates, [])} | Acc], Timeout);
 update_item(T, K, [{expected, V}|Rest], Acc, Timeout) ->
-    update_item(T, K, Rest, [{<<"Expected">>, attr_updates(V, [])}|Acc], Timeout);
+    update_item(T, K, Rest, [{<<"Expected">>, attr_updates(V, [])} | Acc], Timeout);
 update_item(T, K, [{return, none}|Rest], Acc, Timeout) ->
-    update_item(T, K, Rest, [{<<"ReturnValues">>, ?NONE}|Acc], Timeout);
+    update_item(T, K, Rest, [{<<"ReturnValues">>, ?NONE} | Acc], Timeout);
 update_item(T, K, [{return, all_old}|Rest], Acc, Timeout) ->
-    update_item(T, K, Rest, [{<<"ReturnValues">>, ?ALL_OLD}|Acc], Timeout);
+    update_item(T, K, Rest, [{<<"ReturnValues">>, ?ALL_OLD} | Acc], Timeout);
 update_item(T, K, [{return, updated_old}|Rest], Acc, Timeout) ->
-    update_item(T, K, Rest, [{<<"ReturnValues">>, ?UPDATED_OLD}|Acc], Timeout);
+    update_item(T, K, Rest, [{<<"ReturnValues">>, ?UPDATED_OLD} | Acc], Timeout);
 update_item(T, K, [{return, all_new}|Rest], Acc, Timeout) ->
-    update_item(T, K, Rest, [{<<"ReturnValues">>, ?ALL_NEW}|Acc], Timeout);
+    update_item(T, K, Rest, [{<<"ReturnValues">>, ?ALL_NEW} | Acc], Timeout);
 update_item(T, K, [{return, updated_new}|Rest], Acc, Timeout) ->
-    update_item(T, K, Rest, [{<<"ReturnValues">>, ?UPDATED_NEW}|Acc], Timeout).
+    update_item(T, K, Rest, [{<<"ReturnValues">>, ?UPDATED_NEW} | Acc], Timeout).
 
 %% query_item options:
+%% Uses amazon api version: http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/API_Query_v20111205.html
 %% limit: int, max number of results
 %% count: bool, only return the total count
 %% scan_index_forward: bool, set to false to reverse the sort order
 %% consistent: bool, make a consistent read, default false
 %% exclusive_start_key: output from LastEvaluatedKey when limit(size or limit param) is reached
 %% attrs: array( binary ), [<<"a">>,<<"b">>], only return these attributes
 %% range_condition: {array( attributes), operation }
-%% eg, dinerl:query_item(<<"table">>,[{<<"S">>, <<"hash_value">>}], [{range_condition, { [[{<<"S">>, <<"range_value">>}]]  ,<<"EQ">> }}]).
-
+%% eg, dinerl:query_item(<<"table">>,[{<<"S">>, <<"hash_value">>}],
+%%   [{range_condition, { [[{<<"S">>, <<"range_value">>}]]  ,<<"EQ">> }}]).
 query_item(Table, Key, Options) ->
     query_item(Table, Key, Options, undefined).
 query_item(Table, Key, Options, Timeout) ->
     query_item(Table, Key, Options, [], Timeout).
 
 query_item(T, K, [], Acc, Timeout) ->
```

Comment:
We are missing the no base cases. Without them FBX will burn down because of `ProductActions` queries.